### ISSUE TITLE
High idle CPU usage with STT on Rust server

### ISSUE THREAD

**Issue by coder2543** [kyutai-labs/moshi#390]
### Backend impacted

The Rust implementation

### Operating system

Linux

### Hardware

GPU with CUDA

### Description

Good day,

Getting high CPU usage when running Rust server for STT, even when no requests being processed. Running latest version 0.6.4

Relevant bug in Rust server for TTS, which has been fixed: https://github.com/kyutai-labs/moshi/issues/363 

Command: moshi-server worker --config configs/stt-prod.toml

(stt-prod.toml is here: https://github.com/kyutai-labs/unmute/blob/main/services/moshi-server/configs/stt-prod.toml)

The issue is correlated with batch size:
When batch size is set to 8 in stt.toml, 500% CPU usage on idle (5 CPU threads utilized).
When it is set to 64, 1200% CPU usage on idle. (12 CPU threads fully utilized).

The STT model is loaded in GPU memory, and GPU utilization is 0% when no requests are being processed, as expected.

Environment: Linux, modern AMD CPU, modern NVIDIA GPU with CUDA 12

Thank you.

### Extra information

Command: moshi-server worker --config configs/stt-prod.toml

(stt-prod.toml is here: https://github.com/kyutai-labs/unmute/blob/main/services/moshi-server/configs/stt-prod.toml)

Similar bug in Rust server for TTS, which has been fixed in version 0.6.4: https://github.com/kyutai-labs/moshi/issues/363 

---
stt-prod.toml:

static_dir = "./static/"
log_dir = "/tmp/unmute_logs"
instance_name = "tts"
authorized_ids = ["public_token"]

[modules.asr]
path = "/api/asr-streaming"
type = "BatchedAsr"
lm_model_file = "hf://kyutai/stt-1b-en_fr-candle/model.safetensors"
text_tokenizer_file = "hf://kyutai/stt-1b-en_fr-candle/tokenizer_en_fr_audio_8000.model"
audio_tokenizer_file = "hf://kyutai/stt-1b-en_fr-candle/mimi-pytorch-e351c8d8@125.safetensors"
asr_delay_in_tokens = 6
batch_size = 64  # NOTE: make this smaller if running on one GPU
conditioning_learnt_padding = true
temperature = 0.25

[modules.asr.model]
audio_vocab_size = 2049
text_in_vocab_size = 8001
text_out_vocab_size = 8000
audio_codebooks = 20

[modules.asr.model.transformer]
d_model = 2048
num_heads = 16
num_layers = 16
dim_feedforward = 8192
causal = true
norm_first = true
bias_ff = false
bias_attn = false
context = 750
max_period = 100000
use_conv_block = false
use_conv_bias = true
gating = "silu"
norm = "RmsNorm"
positional_embedding = "Rope"
conv_layout = false
conv_kernel_size = 3
kv_repeat = 1
max_seq_len = 40960

[modules.asr.model.extra_heads]
num_heads = 4
dim = 6

### Environment

Fill in the following information on your system.
- Operating system version:

If the backend impacted is PyTorch:
- Python version:
- PyTorch version:
- CUDA version (run `python -c 'import torch;  print(torch.version.cuda)'`):
- GPU model and memory:
  
If the backend is MLX:
- Mac model:

**Comment by gabrieldemarmiesse** [comment]
Hello @coder2543, 

I've had a similar issue a while back and fixed it by using this environment variable: `RAYON_NUM_THREADS=4`. Can you try it out?
